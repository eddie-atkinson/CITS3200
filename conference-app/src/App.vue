<template>
  <v-app>
    <v-app-bar
      app
      color="primary"
      dark
    >
      <v-btn
      text
      @click='$vuetify.theme.dark=!$vuetify.theme.dark'
      >
        <v-icon
        v-show='!$vuetify.theme.dark'
        >
          mdi-lightbulb-cfl-off
        </v-icon>
        <v-icon
        v-show='$vuetify.theme.dark'
        >
          mdi-lightbulb-cfl
        </v-icon>
      </v-btn>
      <v-spacer></v-spacer>
      <v-btn
        to="/"
        text
      >
        Start over
      </v-btn>
    </v-app-bar>
    <v-main>
      <router-view>
      </router-view>
    </v-main>
  </v-app>
</template>

<script>

export default {
  name: 'App',

  components: {
  },

  data: () => ({
  }),
  methods: {
  },
};
</script>
