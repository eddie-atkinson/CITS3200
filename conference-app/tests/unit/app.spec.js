import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import Vue from 'vue';
import App from '@/App.vue'; // page being tested

Vue.use(Vuetify);

const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

describe('App', () => {
  it('has data', () => {
    expect(typeof App.data).toBe('function');
  });
});

describe('Mounted App', () => {
  let vuetify;

  // Before each test we should ...
  beforeEach(() => {
    vuetify = new Vuetify(); // Instantiate a new Vue instance
  });

  test('is instantiated', () => { // App.vue correctly mounts
    const wrapper = shallowMount(App, {
      localVue,
      vuetify,
      router,
    });

    expect(wrapper.vm).toBeTruthy();
  });

  // UNSURE HOW TO GET THIS TO WORK FOR NOW

  /*
  it('has correct title', () => {
    const titleWrapper = wrapper.find('title');
    expect(titleWrapper.text()).toEqual('Conference Converter - Home');
  });
  */

  // currently this will not work due to start over button and intro card in diff views
  // perhaps this is best suited as an e2e test
  /*
  it('starts over correctly when pressed from landing page', async () => {
    const wrapper = mount(App, {
      localVue,
      vuetify,
      router,
    });

    const restartButton = wrapper.find('.v-btn--router');
    const vcardWrappers = wrapper.findAll('v-card-text');
    expect(vcardWrappers.exists()).toBe(true);

    await restartButton.trigger('click');

    expect(vcardWrappers[0]).toContain('Welcome to the Australian');
    expect(vcardWrappers[1]).toContain('To get started');
  });
  */

  it('switches between light/dark mode', async () => {
    const wrapper = mount(App, {
      localVue,
      vuetify,
      router,
    }); // mount() includes child elements

    const bodyWrapper = wrapper.find('div'); // colour scheme is applied to the first div of the document

    const darkModeButton = wrapper.find('.v-btn');
    expect(darkModeButton.exists()).toBe(true); // we can find our button correctly

    await darkModeButton.trigger('click');
    expect(bodyWrapper.classes()).toContain('theme--dark');
    await darkModeButton.trigger('click');
    expect(bodyWrapper.classes()).toContain('theme--light');
  });

  /*
  it('correctly renders excel spreadsheet', () => {

  });
  */

  /*
    Snapshot testing, will initialise snapshot upon first run and all following
    If snapshot was intended to be modified, snapshot can be updated with npm run test:unit -- -u
  */

  it('matches last snapshot', () => {
    const wrapper = shallowMount(App, {
      localVue,
      vuetify,
      router,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
