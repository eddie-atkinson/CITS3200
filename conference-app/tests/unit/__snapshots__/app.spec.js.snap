// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App matches last snapshot 1`] = `
<v-app-stub id="app">
  <v-app-bar-stub color="primary" dark="true" tag="header" extensionheight="48" src="" value="true" app="true">
    <v-btn-stub tag="button" activeclass="" text="true" type="button">
      <v-icon-stub>
        mdi-lightbulb-cfl-off
      </v-icon-stub>
      <v-icon-stub>
        mdi-lightbulb-cfl
      </v-icon-stub>
    </v-btn-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-btn-stub tag="button" activeclass="" to="/" text="true" type="button">
      Start over
    </v-btn-stub>
  </v-app-bar-stub>
  <v-main-stub tag="main">
    <router-view-stub name="default"></router-view-stub>
  </v-main-stub>
</v-app-stub>
`;
